### PR TITLE
Adding dual stack IPv4 + IPv6 support to DHT.

### PR DESCRIPTION
**Pull request by spikebike**
Open to suggestions on how to improve the code or organize it better welcome.

It's working well for me.  Here's a run:
$ go run main6.go -v6 [2601:c:a200:1228:76d0:2bff:fe90:8b90] 12b9b08b325e877d08cc63a75e8fd24699c3dc6b
Tring to bind to IPv6=[2601:c:a200:1228:76d0:2bff:fe90:8b90]
=========================== DHT
Note that there are many bad nodes that reply to anything you ask.
Peers found:
=========================== DHT
Note that there are many bad nodes that reply to anything you ask.
Peers found:
0: [2505:20f6:19c2:2a:9c01:fe:00ca:dd11]:8445
0: 114.40.158.84:19944
1: 220.135.171.68:11183
2: 125.81.132.91:8978
3: 154.45.216.164:1073
4: 185.6.124.193:49572
1: [2601:c:a200:1228:76d0:2bff:fe90:8b90]:8445

### CODE REVIEW

File: dht.go [hunk 1]
```diff
@@ -84,15 +84,17 @@ type Config struct {
...
 
 // Creates a *Config populated with default values.
 func NewConfig() *Config {
 	return &Config{
 		Address:          "",
-		Port:             0, // Picks a random port.
+		Port:             8444, // Picks a random port.
 		NumTargetPeers:   5,
-		DHTRouters:       "de.magnets.im:6881,cz.magnets.im:6881,router.bittorrent.com:6881,dht.transmissionbt.com:6881",
+		DHTRouters:       "[2001:41d0:1:aa81:2::1]:6881",
```
Comment (spikebike) [inline dht.go]: The IPv6 IP for the bootstrap node was dht.transmissionbt.com which has IPv4 and IPv6 (A and AAAA) records.  I returned the list to the popular bootstrap nodes.

**Comment by spikebike** [comment]
This should go in much cleaner.  As requested:
* replaced all import references back to original
* used DNS instead of IP addresses for bootstrap, list includes IPv4 and IPv6 bootstrap nodes
* added main6.go example, it launches an IPv4 and IPv6 DHT to find the infohash.
* tested result with go-1.4

Any and all suggestions welcome, if you prefer I create a new pull request just let me know.

**Comment by nictuku** [comment]
This looks great. Thank you Bill.

**Comment by spikebike** [comment]
"This looks great. Thank you Bill."  Doesn't look like the pull was accepted.  Need anything else from me?

**Comment by nictuku** [comment]
bah, I could swear I had merged it. Looks like there are conflicts to resolve. I'll do that ASAP

File: dht.go [hunk 2]
```diff
@@ -100,6 +102,7 @@ func NewConfig() *Config {
 		RateLimit:        100,
 		MaxInfoHashes:    2048,
 		MaxInfoHashPeers: 256,
+		UDPProto:         "udp6",
```
Comment (nictuku) [inline dht.go]: I've changed this to udp4 as the default. I hope that's OK.

File: routing_table.go [hunk 3]
```diff
@@ -7,6 +7,7 @@ import (
 	"time"
 
 	log "github.com/golang/glog"
+	"github.com/spikebike/nettools"
```
Comment (nictuku) [inline routing_table.go]: I've changed this to nictuku/nettools. I hope that doesn't break your change.

When doing changes like this, use the original repository in the import line - go get the original repo, then add your fork as a remote so you can work locally with those changes.

Comment (spikebike) [inline routing_table.go]: On 01/24/2015 04:37 PM, Yves Junqueira wrote:
> In routing_table.go
> <https://github.com/nictuku/dht/pull/32#discussion_r23500747>:
> 
>> @@ -7,6 +7,7 @@ import (
>>  	"time"
>>  
>>  	log "github.com/golang/glog"
>> +	"github.com/spikebike/nettools"
> 
> I've changed this to nictuku/nettools. I hope that doesn't break your
> change.

Ugh, I thought I fixed those, I guess I missed one.

> When doing changes like this, use the original repository in the import
> line - go get the original repo, then add your fork as a remote so you
> can work locally with those changes.

Not sure exactly how to do that actually, but I did intend to revert
that change before issuing the pull.

**Pull request merged**